### PR TITLE
Support logger stopping at std::exit

### LINKED ISSUE #9720: PR builds warning cancelation causes background lock
## Quick summary
image
Pressing `No` in this warning window hangs `rpcs3` in background with 20% CPU usage. I suppose something with busy waiting.
Sorry for non-English screenshot.
image

## Details

Waited for 5 mins, still stuck in background.

#### 1. Please attach RPCS3's log.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/5929495/RPCS3.log)
Obtained after killing process in Task Manager
Tested with #9717.

#### 5. Please provide your system configuration:
Test purposes laptop (irrelevant for the issue).
* Win 10
* Core I5 10210U
* NVidia MX250 2G
* Driver 460.79.0.0
* etc.

#### Please include.
Any additionals test for you is a pleasure

### PR DESCRIPTION
**Pull request by elad335**
Fixes #9720 , alternative to #9721.
Registers destructors for std::exit as if by std::atexit.

### CODE REVIEW

**Comment by elad335** [comment]
More static at main()

**Comment by elad335** [comment]
There are probably more variables needing static for various reasons, please note them in pr comments. This pr is not linked to the mentioned issue exclusively of course.

**Comment by Megamouse** [comment]
I would be more interested in a proper explanation. Not only here but also as comment in the code.
Changes like these seem more like a workaround and are too easy to break if new stuff is added.

**Comment by elad335** [comment]
Added description.

File: rpcs3/main.cpp [hunk 1]
```diff
@@ -451,12 +450,12 @@ int main(int argc, char** argv)
 	// but I haven't found an implicit way to check for style yet, so we naively check them both here for now.
 	const bool use_cli_style = find_arg(arg_style, argc, argv) || find_arg(arg_stylesheet, argc, argv);
 
-	QScopedPointer<QCoreApplication> app(createApplication(argc, argv));
+	static QScopedPointer<QCoreApplication> app(createApplication(argc, argv));
```
Comment (Nekotekina) [inline rpcs3/main.cpp:453]: I think normal application should not outlive main()

Comment (elad335) [reply rpcs3/main.cpp]: Fixed.

**Pull request merged**